src/game/rendering_graph_node.c

```diff
@@ -248,6 +252,83 @@ Mtx identityMatrixWorldScale = {{
      0x00000000,                            LOWER_FIXED(1.0f)               <<  0}
 }};
 
+static void lists_render(Gfx **ptempGfxHead, struct DisplayListNode* currList)
+{
+#define tempGfxHead (*ptempGfxHead)
+    do {
+        gSPMatrix(tempGfxHead++, VIRTUAL_TO_PHYSICAL(currList->transform), (G_MTX_MODELVIEW | G_MTX_LOAD | G_MTX_NOPUSH));
+        gSPDisplayList(tempGfxHead++, currList->displayList);
+        currList = currList->next;
+    } while (currList != NULL);
+#undef tempGfxHead
+}
+
+static int batches_render(Gfx **ptempGfxHead, struct BatchArray* arr, u32 mode1, u32 mode2)
+{
+#define tempGfxHead (*ptempGfxHead)
+    int amountRendered = 0;
+
+    // Some "fun" display lists before may decide to change the render mode so we need to reset it.
+    // Notably this happens when env effects that change render mode and do not revert it.
+    gDPSetRenderMode(tempGfxHead++, mode1, mode2);
+
+    for (int batch = 0; batch < arr->count; batch++) {
+        struct Batch* batches = &arr->batches[batch];
+        if (!batches->list.head)
+            continue;
+
+        gSPDisplayList(tempGfxHead++, batches->startDl);
+        amountRendered++;
+        lists_render(&tempGfxHead, batches->list.head);
+        gSPDisplayList(tempGfxHead++, batches->endDl);
+    }
+#undef tempGfxHead
+
+    return amountRendered;
+}
+
+static void main_render(Gfx **ptempGfxHead, struct DisplayListNode* currList, u32 mode1, u32 mode2, int phaseIndex)
+{
+    (void)phaseIndex;
```

Comment:
Is this a leftover compiler warning bypass?

Reply:
Yes, I prefer to be explicit here when silhouette is off. I have moved it under SILHOUETTE.

Reply:
Can you leave a short comment here too just to clarify it's a compiler warning bypass? You could also use the UNUSED keyword (though this is still messy)